### DISCUSSION TITLE
RNode connecting to reticulum via GPRS/LTE modem

### DISCUSSION THREAD

**Discussion by wzab** [markqvist/Reticulum#356] [Ideas]
Hi All,

I'd like to create a small LoRa-based Reticulum network connected via the Internet to the rest of my Reticulum network.
Of course, I can do it by connecting an RNode to the Raspberry Pi, which is connected via a GPRS/LTE modem to the Internet and runs the Reticulum stack.
However, due to the power supply limitations, I'd prefer a solution where the RNode is equipped with the GPRS/LTE modem (e.g., based on SIM800L or SIM7600G, boards TTGO T-Call or TTGO T-SIM7600E-L1C), and connects via the Internet to my Reticulum gateway. Is such a solution easily implementable in the current RNode firmware?

BR, Wojtek

**Comment by Erethon** [comment]
Hello, if I understand correctly you essentially want to get rid of the Raspberry in the setup for power consumption reasons. Unfortunately, for now Reticulum requires Python in order to work, so you can't run it standalone on a microcontroller, you need a "computer" to run Reticulum. #21 is the discussion where people are discussing the work needed to port Reticulum so it's possible to run it standalone on a MCU.

**Reply by wzab** [reply]
Hello, I thought that maybe it could be possible to run the Reticulum stack on a remote Linux machine, to which the RNode connects via a cellular modem. Just by replacing a USB connection with something tunneled via a TCP or UDP link.

**Reply by faragher** [reply]
Let me see if I understand your use case. You want an RNode to connect to your cellular network to transmit to a remote Linux machine that's running the full Reticulum stack.

Basically you wish to tunnel what would be transferred over serial over the Internet using a cellular connection? You'd then use the RNode with cellular data as a local radio for others to link into? Am I understanding that correctly?

The devil is in the timing details, but it seems feasible. I haven't done it, and there's always the risk of libraries getting too large, but you might be able to put it all on one board with custom hardware.

The other three options are to get a small cellular modem board and make a custom firmware, get a cellular board with a microprocessor and use that to send the serial information, or get a full fledged remote terminal unit for a basestation. 

In any case, you'd need to find a good way to link the system to Reticulum. I'd suggest writing a program that presents itself as a serial device and sends/receives over your chosen uplink. (likely just a TCP socket) Then Reticulum can just run as normal, timings permitting.

There are a lot of moving parts, and it's not a trivial thing to do, but it's fascinating and allows for a simpler endpoint.

However, bear in mind that each and every piece of data, even between RNodes in the area, will be sent across this link. It may be much, much slower and robust than you'd hope, especially in areas with poor cellular reception.

**Reply by faragher** [reply]
Addendum: It appears socat can create a virtual com port over TCP, so the Linux side is basically all stock.
